app/views/advertisements/show.js.erb

```diff
@@ -1,19 +1,37 @@
 (function() {
+  function polyfillCustomEvent() {
+    if ( typeof window.CustomEvent === "function" ) return false;
+    function CustomEvent ( event, params ) {
+      params = params || { bubbles: false, cancelable: false, detail: undefined };
+      var evt = document.createEvent( 'CustomEvent' );
+      evt.initCustomEvent( event, params.bubbles, params.cancelable, params.detail );
+      return evt;
+    }
+    CustomEvent.prototype = window.Event.prototype;
+    window.CustomEvent = CustomEvent;
+  }
```

Comment:
The polyfill adds some unnecessary bloat. I'd opt for simply creating the event like so.

```javascript
var evt = new Event("codefund");
evt.detail = { status: 'ok' };
window.dispatchEvent(evt);
```

https://developer.mozilla.org/en-US/docs/Web/API/Event/Event